packages/envconfig/src/envconfig-toml.ts

```diff
@@ -0,0 +1,326 @@
+import { readFileSync } from 'fs';
+import * as path from 'path';
+import * as os from 'os';
+import { parse, stringify, TomlTable } from 'smol-toml';
+import {
+  ConfigDataSource,
+  LoadClientConfigOptions,
+  LoadClientProfileOptions,
+  TomlClientConfig,
+  TomlClientConfigCodec,
+  TomlClientConfigProfile,
+  TomlClientConfigTLS,
+} from './types';
+import { normalizeGrpcMetaKey } from './utils';
+
+function sourceToStringData(source?: ConfigDataSource): string | undefined {
+  if (source === undefined) {
+    return undefined;
+  }
+  if ('path' in source) {
+    return readFileSync(source.path)?.toString();
+  }
+
+  if (Buffer.isBuffer(source.data)) {
+    return source.data.toString();
+  }
+  return source.data;
+}
+
+type EnvProvider = { kind: 'Map'; map: Record<string, string> } | { kind: 'System' };
+
+function getEnvVar(provider: EnvProvider, key: string): string | undefined {
+  if (provider.kind === 'Map') {
+    return provider.map[key];
+  }
+  return process.env[key];
+}
+
+export function tomlLoadClientConfig(options: LoadClientConfigOptions): TomlClientConfig {
+  const envProvider: EnvProvider = options.overrideEnvVars
+    ? { kind: 'Map', map: options.overrideEnvVars }
+    : { kind: 'System' };
+
+  let configData = undefined;
+  try {
+    configData = sourceToStringData(options.configSource) ?? getFallbackConfigData(envProvider);
```

Comment:
Are you sure about having the fallback in the try block? You say below that File Not Found is ok… Shouldn't we default to fallback config in that case?

Reply:
Hmm, valid but we haven't done this in other SDKs. The core/go/java approach falls back to empty profile objects if the file isn't found:

Ex (core)
```
    if let Some(data) = toml_data {
        ClientConfig::from_toml(
            &data,
            ClientConfigFromTOMLOptions {
                strict: options.config_file_strict,
            },
        )
    } else {
        Ok(ClientConfig::default())
    }
```